components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Extensions/PublishedDataSetSourceModelEx.cs

```diff
@@ -42,11 +42,13 @@ public static class PublishedDataSetSourceModelEx {
                 model.Connection?.User?.Type.ToString() +
                 model.Connection?.User?.Value.ToJson() +
                 model.SubscriptionSettings?.PublishingInterval.ToString() +
-                model.PublishedVariables.PublishedData.First()?.Id +
-                model.PublishedVariables.PublishedData.First()?.PublishedVariableNodeId +
-                model.PublishedVariables.PublishedData.First()?.PublishedVariableDisplayName +
-                model.PublishedVariables.PublishedData.First()?.SamplingInterval +
-                model.PublishedVariables.PublishedData.First()?.HeartbeatInterval;
+                model.PublishedVariables.PublishedData.FirstOrDefault()?.Id +
```

Comment:
please optimize and store the first PublishedData into local variable

Reply:
Fixed it.